nbodykit/base/tests/test_catalog.py

```diff
@@ -1,12 +1,59 @@
 from runtests.mpi import MPITest
 from nbodykit.lab import *
-from nbodykit import setup_logging
+from nbodykit import setup_logging, set_options
 
 import pytest
 from numpy.testing import assert_allclose, assert_array_equal
 
 setup_logging()
 
+
+@MPITest([1, 4])
+def test_consecutive_selections(comm):
+
+    CurrentMPIComm.set(comm)
+
+    # compute with smaller chunk size to test chunking
+    with set_options(dask_chunk_size=100):
+
+        s = UniformCatalog(1000, 1.0, seed=42)
+
+        # slice once
+        subset1 = s[:20]
```

Comment:
Is this :20 per rank or :20 of the entire dataset?

Reply:
Maybe we shall define gslice(a, b, c) that represents global selection cross ranks? It would be useful in abundance matching a sorted catalog.

Reply:
Yes, currently the slicing is all local. A global slice would be nice. I think I have some code I started on for that already. I'll open up a new PR for it